march_hardware/src/EtherCAT/EthercatMaster.cpp

```diff
@@ -217,4 +219,19 @@ void EthercatMaster::stop()
   }
 }
 
+void EthercatMaster::setThreadPriority(int priority)
+{
+  struct sched_param param = { priority };
+  // SCHED_FIFO scheduling preempts other threads with lower priority as soon as it becomes runnable.
+  // See http://man7.org/linux/man-pages/man7/sched.7.html for more info.
+  const int error = pthread_setschedparam(this->ethercat_thread_.native_handle(), SCHED_FIFO, &param);
+  if (error != 0)
+  {
+    ROS_ERROR("Failed to set the ethercat thread priority to %d. (error code: %d)", priority, error);
+  }
+  else
+  {
+    ROS_DEBUG("Set ethercat thread priority to %d", param.sched_priority);
```

Comment:
Do we ever look at debugs? They don't get logged either right?

Reply:
Not by default no, but if you want to you can :point_right: